Core/include/Acts/Geometry/detail/NavigationStateUpdators.hpp

```diff
@@ -0,0 +1,173 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Definitions/Common.hpp"
+#include "Acts/Geometry/NavigationDelegates.hpp"
+#include "Acts/Utilities/BinningType.hpp"
+#include "Acts/Utilities/Helpers.hpp"
+
+#include <array>
+#include <memory>
+
+namespace Acts {
+namespace Experimental {
+namespace detail {
+
+/// @brief  This sets a single object, e.g. single surface or single volume
+/// @tparam object_type the type of the object to be filled
+/// @tparam filler_type is the helper to fill the object into nState
+template <typename object_type, typename filler_type>
+class SingleObjectImpl : public IManagedDelegateImpl {
+ public:
+  // The object to be filled in
+  const object_type* object = nullptr;
+
+  /// Convenience constructor
+  /// @param so the single object
+  SingleObjectImpl(const object_type* so) : object(so) {}
+
+  /// @brief updates the navigation state with a single object that is filled in
+  ///
+  /// @param gctx is the Geometry context of this call
+  /// @param nState the navigation state to which the surfaces are attached
+  ///
+  /// @note this is attaching objects without intersecting nor checking
+  void update([[maybe_unused]] const GeometryContext& gctx,
+              NavigationState& nState) const {
+    filler_type::fill(nState, object);
```

Comment:
Ah, that's why you need the struct. This is an implicit concept here? Do I get this right, @asalzburger?

Reply:
As you can see, this is one way of implementing the `NavigationStateUpdator` delegate, that's why it's also in `detail`, within here I use some concepts, but it is internally to that implementation.

Reply:
We will do this, once you have a new `ManagedDelegate<>` submitted.